### PR TITLE
chore: Update SETUP.md to include part about format on save

### PR DESCRIPTION
**Pull request by AliceR**
**Related Ticket:**  #1263

### Description of Changes
Developers should configure their IDEs or text editors to automatically format code on save.
This ensures code is consistently styled as per project standards before it even reaches version control. 

This adds documentation in SETUP.md on how to configure format-on-save for VS Code.

### Validation / Testing
Follow the steps described in the SETUP.md. Confirm that everything formats as expected.

### CODE REVIEW

File: docs/development/SETUP.md [hunk 1]
```diff
@@ -62,6 +62,24 @@ VEDA-UI includes a GitHub action for checking TypeScript and lint errors. If you
...
 Additionally, there's a pre-commit hook that performs the same error checks. This helps developers identify and address issues at an earlier stage. If you need to bypass the check (to make a local temporary commit etc.), include the `--no-verify`` flag in your commit command.
 
+#### Format on Save
+While the existing checks ensure that no linting errors are introduced into the main branch, enabling "Format on Save" in your code editor provides an additional layer of convenience. It helps catch formatting issues early in the development process, minimizing the risk of committing errors and ensuring consistent code style across the team.
+
+##### How to Set Up Format on Save in VS Code
+Install Required Extensions:
+
+1. Install the ESLint extension ("dbaeumer.vscode-eslint").
+2. Enable Format on Save in the settings (Cmd+, on macOS)
```
Comment (hanbyul-here) [inline docs/development/SETUP.md:72]: Is the shortcut for settings `Cmd+` ? Mine shows that it is `Cmd,` !
Screenshot 2024-12-11 at 11 28 58 AM

Comment (AliceR) [reply docs/development/SETUP.md:72]: That actually means `Cmd` + `,`, so the same as yours. 
This should make it easier to read:
```suggestion
2. Enable Format on Save in the settings (<kbd>⌘ Command</kbd> + <kbd>,</kbd> on macOS)
```

Comment (dzole0311) [reply docs/development/SETUP.md:72]: That looks great!

Screenshot 2024-12-12 at 10 11 53

Comment (hanbyul-here) [reply docs/development/SETUP.md:72]: ohh haha, sorry for. misreading! Thanks for making it clear!

**Review by vgeorge — APPROVED** [review]

**Review by snmln — APPROVED** [review]

File: docs/development/SETUP.md [hunk 2]
```diff
@@ -70,6 +70,24 @@ VEDA-UI includes a GitHub action for checking TypeScript and lint errors. If you
...
+##### How to Set Up Format on Save in VS Code
+Install Required Extensions:
+
+1. Install the ESLint extension ("dbaeumer.vscode-eslint").
+2. Enable Format on Save in the settings (<kbd>⌘ Command</kbd> + <kbd>,</kbd> on macOS)
+3. Configure Formatter Settings:
+   Ensure ESLint is set as the default formatter:
+   ```json
+   "editor.defaultFormatter": "dbaeumer.vscode-eslint"
+   ```
```
Comment (sandrahoang686) [inline docs/development/SETUP.md:85]: was able to sortof follow.. following step 2, couldn't figure out where to add snippet so just opened command palette with cmd+shift+p and found `Open User Settings (json)` and then added snippet.

Also must be a small type but i dont think `json` is meant to be after the "```"

File: docs/development/SETUP.md [hunk 3]
```diff
@@ -70,6 +70,24 @@ VEDA-UI includes a GitHub action for checking TypeScript and lint errors. If you
...
+5. Test the Configuration:
+   Make a change to any file in your project, save it, and verify that it is automatically formatted.
+
+Note: The project contains a file `.vscode/settings.json.sample` with our recommended vscode settings.
+
```
Comment (sandrahoang686) [inline docs/development/SETUP.md:90]: Is this `Note` a heads up that every project should contain that file with the recommended settings? Or just telling the user what already exists?

Comment (AliceR) [reply docs/development/SETUP.md:90]: This is just to tell the user this file exists. 
```suggestion
To apply these settings to your workspace in VS Code, rename the file to `.vscode/settings.json` by removing the `.sample` extension.
```

File: docs/development/SETUP.md (hunk 2 above)
Comment (AliceR) [inline docs/development/SETUP.md:85]:
```suggestion
   Set ESLint as the default formatter:

   **Using Settings UI**: Open the Command Palette (Cmd+Shift+P or Ctrl+Shift+P), select Preferences: Open Settings (UI), search "default formatter," and select ESLint.

   **Using JSON**: Open Preferences: Open User Settings (JSON) and add:
   
   ```
   "editor.defaultFormatter": "dbaeumer.vscode-eslint"
   ```
```

Comment (AliceR) [reply docs/development/SETUP.md:85]: Thanks @sandrahoang686 ! Do you think adding more detailed instructions would help? I did not want to go too much into vs code specifics, as people may be using different editors. Maybe with the updated note below, we don't need to change anything here at all? Let me know what you think!

Comment (sandrahoang686) [reply docs/development/SETUP.md:85]: yeah I think that helps! thanks for working on this & updating 🙇🏼‍♀️

**Review by sandrahoang686 — APPROVED** [review]

File: docs/development/SETUP.md [hunk 4]
```diff
@@ -70,6 +70,25 @@ VEDA-UI includes a GitHub action for checking TypeScript and lint errors. If you
...
+#### Format on Save
+While the existing checks ensure that no linting errors are introduced into the main branch, enabling "Format on Save" in your code editor provides an additional layer of convenience. It helps catch formatting issues early in the development process, minimizing the risk of committing errors and ensuring consistent code style across the team.
+
+##### How to Set Up Format on Save in VS Code
+Install Required Extensions:
+
+1. Install the ESLint extension ("dbaeumer.vscode-eslint").
+2. Enable Format on Save in the settings (<kbd>⌘ Command</kbd> + <kbd>,</kbd> on macOS)
+3. Configure Formatter Settings:
+   Ensure ESLint is set as the default formatter:
```
Comment (AliceR) [inline docs/development/SETUP.md:82]:
```suggestion
   Ensure ESLint is set as the default formatter:
      **Using Settings UI**: Open the Command Palette (Cmd+Shift+P or Ctrl+Shift+P), select Preferences: Open Settings (UI), search "default formatter," and select ESLint.

   **Using JSON**: Open Preferences: Open User Settings (JSON) and add:
```

File: docs/development/SETUP.md [hunk 5]
```diff
@@ -70,6 +70,28 @@ VEDA-UI includes a GitHub action for checking TypeScript and lint errors. If you
...
+      **Using Settings UI**: Open the Command Palette (Cmd+Shift+P or Ctrl+Shift+P), select Preferences: Open Settings (UI), search "default formatter," and select ESLint.
```
Comment (AliceR) [inline docs/development/SETUP.md:83]:
```suggestion
      **Using Settings UI**: Open the Command Palette (<kbd>⌘ Command</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd> on macOS), select Preferences: Open Settings (UI), search "default formatter," and select ESLint _dbaeumer.vscode-eslint_.
```

**Pull request merged**